sdk/go/pulumi-language-go/main.go

```diff
@@ -1169,14 +1169,16 @@ func (host *goLanguageHost) GeneratePackage(
 			Diagnostics: rpcDiagnostics,
 		}, nil
 	}
-	files, err := codegen.GeneratePackage("pulumi-language-go", pkg)
+	files, err := codegen.GeneratePackage("pulumi-language-go", pkg, req.LocalDependencies)
 	if err != nil {
 		return nil, err
 	}
 
 	for filename, data := range files {
 		outPath := filepath.Join(req.Directory, filename)
-
+		if filename != "go.mod" && req.LocalDependencies != nil {
+			outPath = filepath.Join(req.Directory, "go", filename)
+		}
```

Comment:
Unfortunate that this is required 😞 everything generated needs to be under `go` directory, except for the top-level go.mod file. We check that `req.LocalDependencies != nil` to maintain current behavior in case people use `gen-sdk`

Reply:
I'm inclined to go with we take a break change here and just make the module just be the go folder, i.e. don't do this nesting in codegen write the code and mod to the output folder (which will generally be "go"), new providers can have nice clean import paths. Old providers can hack fix it by generating the sdk and then moving the go.mod up one folder.

Reply:
> I'm inclined to go with we take a break change here

We'll need to discuss this. There are too many external customers building providers that we can't pull the rug up from underneath them. To start, it's likely gonna have to be an option that's opt-in.

Reply:
Yes that's why we have the SupportPack option

Reply:
This nesting requires a bigger change because program-gen/sdk-gen assume the `/sdk/go/{pkg}` convention when emitting import paths. It needs to be an option on the schema (for example `sdkRootDir`) which can be `"go"` by default)

Reply:
@Zaid-Ajaj we can remove this check for now. Schema with SupportPack=false will write out exactly the same as today (no subfolder, no go.mod). Schema with SupportPack=true will write out as we expect for conformance tests and for providers going forward (no subfolder, yes go.mod). The expectation of the engine is that the target directory of gen-sdk is a package/module in the target language and that we can immediately pass that folder to Pack to get a releasable artefact.

Reply:
Done ✅